### PR TITLE
YARA Rule importer

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10571)

## Description
A script to import YARA rules into XSOAR.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Yara/ReleaseNotes/1_0_6.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+#### Indicator Types
+
+- **YARA Rule**
```
Comment (ShirleyDenkberg) [inline Packs/Yara/ReleaseNotes/1_0_6.md:4]:
```suggestion
**YARA Rule**
```

File: Packs/Yara/Scripts/ImportYARARule/ImportYARARule.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+args:
+- name: string
+  required: false
+  description: One or more YARA signatures in string format.
+- name: entryID
+  required: false
+  description: An entry ID with for a file containing YARA signatures.
```
Comment (ShirleyDenkberg) [inline Packs/Yara/Scripts/ImportYARARule/ImportYARARule.yml:7]:
```suggestion
  description: An entry ID for a file containing YARA signatures.
```

File: Packs/Yara/Scripts/ImportYARARule/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| Tags | Utility |
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| string | One or more YARA signatures in string format. |
+| entryID | An entry ID with for a file containing YARA signatures. |
```
Comment (ShirleyDenkberg) [inline Packs/Yara/Scripts/ImportYARARule/README.md:17]:
```suggestion
| entryID | An entry ID for a file containing YARA signatures. |
```

**Comment by ShirleyDenkberg** [comment]
@Ni-Knight Doc review completed.

**Review by RotemAmit — COMMENTED** [review]
Nice work!

File: Packs/Yara/Scripts/ImportYARARule/ImportYARARule.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    parser = plyara.Plyara()
+    indicators: list[dict] = []
+    successfully_created_indicators: int = 0
+
+    try:
+        parsed_rules = parser.parse_string(rules)
+        demisto.debug(f'Parsed {len(parsed_rules)} YARA Rules.')
+
+    except plyara.exceptions.ParseError:
+        demisto.debug(f'Failed to parse - {rules}')
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/Yara/Scripts/ImportYARARule/ImportYARARule.py:40]: Done.

File: Packs/Yara/Scripts/ImportYARARule/ImportYARARule.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        if args.get('string', ''):
+            yara_rules: str = args["string"]
+
+        elif args.get('entryID', ''):
+            file_path = demisto.getFilePath(args['entryID'])['path']
+            with open(file_path) as f:
+                yara_rules = f.read()
+
+        else:
+            raise Exception('Input must be provided')
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/Yara/Scripts/ImportYARARule/ImportYARARule.py:131]: Done.

File: Packs/Yara/Scripts/ImportYARARule/ImportYARARule.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+args:
+- name: string
+  required: false
+  description: One or more YARA signatures in string format.
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/Yara/Scripts/ImportYARARule/ImportYARARule.yml:4]: Its not an array, its YARA so no array form needed :)

File: Packs/Yara/ReleaseNotes/1_0_6.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+#### Indicator Types
+
+**YARA Rule**
```
Comment (RotemAmit) [inline Packs/Yara/ReleaseNotes/1_0_6.md:4]:
```suggestion
##### YARA Rule
```

**Review by RotemAmit — COMMENTED** [review]
Please fix the build failures.

File: Packs/Yara/ReleaseNotes/1_0_6.md (hunk 7 above)
Comment (Ni-Knight) [reply Packs/Yara/ReleaseNotes/1_0_6.md:4]: This is the convention for indicator types.

**Review by RotemAmit — APPROVED** [review]
Nice work!

**Pull request merged**